lib/timers.js

```diff
@@ -169,7 +169,11 @@ function listOnTimeout() {
     // Check if this loop iteration is too early for the next timer.
     // This happens if there are more timers scheduled for later in the list.
     if (diff < msecs) {
-      this.start(msecs - diff, 0);
+      var timeRemaining = msecs - (TimerWrap.now() - timer._idleStart);
```

Comment:
@whitlockjc Did you still want to put some of them comments here?

I'd prefer if we tried to make them a bit more concise.. but if that's not possible, it's not a huge deal